uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveBalanceChange.php

```diff
@@ -816,4 +816,51 @@ private static function getBalanceChangesToExpire() {
 
     return $balanceChangesToExpire;
   }
+
+  /**
+   * This method calculates the sum of the TOIL(with the given status) balance changes for a given contact
+   * over a given period of time.
+   *
+   * @param int $contactID
+   * @param DateTime $startDate
+   *   TOILs linked to LeaveRequests with from_date >= this date will be included
+   * @param DateTime $endDate
+   *   TOILs linked to LeaveRequests with with to_date <= this date will be included
+   * @param array $toilStatus
+   *   The statuses of the TOIL request to be included in the calculation/summation
+   *   - An array of values from Leave Request Status option list
+   *
+   * @return float
+   */
+  public static function getTotalTOILBalanceChangeForContact($contactID, DateTime $startDate, DateTime $endDate, $toilStatus = []) {
```

Comment:
This method must also work based on the AbsenceType id. The reason is that we can have multiple absence types which allow TOIL. So, if we have more than one of such absence type, this method will return the balance change sum for all of them, which is not what we need on the toil validation